### DISCUSSION TITLE
Layer model with UI components

### DISCUSSION THREAD

**Discussion by paulorrfilho** [google/filament#3690] [Q&A] [answered]
I am trying to render a model in a fragment and layer it between two UI components (z position). Somehow Filament does not respect the expected layering provided in the activity layout file. 
I tried using the isMediaOverlay attribute in UIHelper but I think it is rendering the model behind my main activity. I also tried changing the elevation in the layout file but it had no impact.
Does Filament support what I am trying to do?

**Comment by romainguy** [comment, marked as answer]
Filament has nothing to do with this. A `SurfaceView` puts its `Surface` either below the `Activity` (by default) or above with `isMediaOverlay`. You cannot sandwich a `SurfaceView` between two UI components. To achieve this you need to use a `TextureView` but it's a little less efficient.

**Reply by paulorrfilho** [reply]
Thanks @romainguy , is this true even if the UI components are in a separate fragments? The Surface view will override any layering I provide?

**Reply by romainguy** [reply]
This has nothing to do with fragments. Think of a `SurfaceView` as a separate *window*.
